pkg/api/folder.go

```diff
@@ -917,19 +853,93 @@ func (fk8s *folderK8sHandler) newToFolderDto(c *contextmodel.ReqContext, item un
 		uid := parentsFullPathUIDs[1:][i]
 		url := dashboards.GetFolderURL(uid, slug)
 
-		parents = append(parents, dtos.Folder{
+		ff := folder.Folder{
 			UID:   uid,
-			OrgID: c.SignedInUser.GetOrgID(),
 			Title: v,
 			URL:   url,
-		})
+		}
+		parentDTO, err := fk8s.toDTO(c, &ff, "", checkCanView)
+		if err != nil {
+			// #TODO should we log this error?
+			return dtos.Folder{}, err
+		}
+
+		parents = append(parents, parentDTO)
 	}
 
 	folderDTO.Parents = parents
 
 	return folderDTO, nil
 }
 
+func toUID(rawIdentifier string) string {
+	// #TODO Is there a preexisting function we can use instead, something along the lines of UserIdentifier?
+	parts := strings.Split(rawIdentifier, ":")
+	if len(parts) < 2 {
+		return ""
+	}
+	return parts[1]
+}
+
+func (fk8s *folderK8sHandler) toDTO(c *contextmodel.ReqContext, fold *folder.Folder, createdBy string, checkCanView bool) (dtos.Folder, error) {
+	// #TODO revisit how/where we get orgID
```

Comment:
I think we can remove this comment
```suggestion
```

Reply:
We still need to figure out which of the parent fields to populate and how to do so.